src/main/java/bio/terra/service/filedata/flight/delete/FileDeleteFlight.java

```diff
@@ -72,7 +72,8 @@ public FileDeleteFlight(FlightMap inputParameters, Object applicationContext) {
     // 4. Delete the directory entry
     // This flight updates GCS and firestore in exactly the reverse order of create, so no new
     // data structure states are introduced by this flight.
-    addStep(new LockDatasetStep(datasetDao, UUID.fromString(datasetId), true), lockDatasetRetry);
+    addStep(
+        new LockDatasetStep(datasetService, UUID.fromString(datasetId), true), lockDatasetRetry);
```

Comment:
We never using the String `datasetId` in the flight, and always do a `UUID.fromString` on it. We should just store the UUID in the flight and pass that to all the usages and not call `UUID.fromString` multiple times.

Reply:
Good point!